File: backend/app/models/task.py
from sqlalchemy import (
    Column,
    Integer,
    String,
    Text,
    DateTime,
    ForeignKey
)
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.database.database import Base


class Task(Base):
    __tablename__ = "tasks"

    id = Column(Integer, primary_key=True, index=True)

    title = Column(String(255), nullable=False)

    description = Column(Text)

    priority = Column(String(20), default="Medium")

    status = Column(String(20), default="Pending")

    deadline = Column(DateTime(timezone=True))

    estimated_time = Column(Integer)

    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now()
    )

    user_id = Column(
        Integer,
        ForeignKey("users.id")
    )

    user = relationship(
        "User",
        back_populates="tasks"
    )

File: frontend/src/types/task.ts
export interface Task {
    id: number;

    title: string;

    description: string;

    priority: "High" | "Medium" | "Low";

    status: "Pending" | "Completed";

    deadline?: string | null;

    estimated_time?: number | null;

    created_at?: string | null;
}

export interface CreateTask {
    title: string;

    description: string;

    priority: "High" | "Medium" | "Low";

    status: "Pending" | "Completed";

    deadline?: string | null;

    estimated_time?: number | null;
}